src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -220,3 +227,178 @@ def test_registration_works_after_failure(self, m: requests_mock.Mocker):
+                        },
+                    },
+                    objecttype_version=3,
+                ),
+                objecttype_url="http://objecttypes-web:8000/api/v2/objecttypes/8e46e0a5-b1b4-449b-b9e9-fa3cea655f48",
+            )
+
+        with self.subTest("Create an object"):
+            submission_create = SubmissionFactory.create(form=form, completed=True)
+            SubmissionStepFactory.create(
+                submission=submission_create,
+                form_step=form_step,
+                data={"voornaam": "My last name"},
+            )
+            object_create_result = plugin.register_submission(
+                submission_create,
+                {
+                    "version": 1,
+                    "objecttype": UUID("8e46e0a5-b1b4-449b-b9e9-fa3cea655f48"),
+                    "objecttype_version": 3,
+                    "objects_api_group": objects_api_group,
+                    "update_existing_object": False,
+                },
+            )
+
+            assert object_create_result is not None
+
+            self.assertNotEqual(object_create_result["uuid"], created_obj)
+            self.assertEqual(
+                object_create_result["record"]["data"]["data"]["fd-test"]["voornaam"],
+                "My last name",
+            )
+
+        with self.subTest("Update existing object"):
+            submission_update = SubmissionFactory.create(
+                form=form,
+                completed=True,
+                initial_data_reference=created_obj["uuid"],
+            )
+            SubmissionStepFactory.create(
+                submission=submission_update,
+                form_step=form_step,
+                data={"voornaam": "Updated value"},
+            )
+            object_update_result = plugin.register_submission(
+                submission_update,
+                {
+                    "version": 1,
+                    "objecttype": UUID("8e46e0a5-b1b4-449b-b9e9-fa3cea655f48"),
+                    "objecttype_version": 3,
+                    "objects_api_group": objects_api_group,
+                    "update_existing_object": True,
+                },
+            )
+
+            assert object_update_result
+
+            with get_objects_client(objects_api_group) as client:
+                obj_retrieved = client.get_object(
+                    object_uuid=object_update_result["uuid"]
+                )
+
+            self.assertEqual(
+                obj_retrieved["record"]["data"]["data"]["fd-test"]["voornaam"],
+                "Updated value",
+            )
+            self.assertEqual(obj_retrieved["uuid"], object_update_result["uuid"])
+
+        with self.subTest(
+            "Existing object not updated when update_existing_object=False"
+        ):
+            submission_update2 = SubmissionFactory.create(
+                form=form,
+                completed=True,
+                initial_data_reference=object_create_result["uuid"],
```

Comment:
```suggestion
                initial_data_reference=created_obj["uuid"],
```
we want to use the object we initially created, right?